libraries/cbor/src/writer.rs

```diff
@@ -56,8 +56,14 @@ impl<'a> Writer<'a> {
                     }
                 }
             }
-            Value::Map(map) => {
-                self.start_item(5, map.len() as u64);
+            Value::Map(mut map) => {
+                map.sort_by(|a, b| a.0.cmp(&b.0));
+                let map_len = map.len();
+                map.dedup_by(|a, b| a.0.eq(&b.0));
```

Comment:
You can use `dedup_by_key(|x| &x.0)`. Same for `sort_by_key`.

Reply:
I was expecting that to work, too. But (see i.e. [here](https://github.com/rust-lang/rust/issues/34162)) there are lifetime issues, therefore the hack.

Reply:
Wow interesting. Wasn't aware of this issue. That's quite problematic.